Makefile

```diff
@@ -271,10 +277,14 @@ install.libseccomp.sudo:
 
 
 cmd/podman/varlink/ioprojectatomicpodman.go: cmd/podman/varlink/io.projectatomic.podman.varlink
+ifeq ($(WANTVARLINK), true)
```

Comment:
Instead of gutting this recipe when `WANTVARLINK` is false, I'd rather adjust the consumers to remove the `cmd/podman/varlink/ioprojectatomicpodman.go` dependency if `WANTVARLINK` is false.

For example, we could have a `VARLINK_GO_DEPS` variable (or some such) that we setup where you're currently setting up `WANTVARLINK`, and then use `VARLINK_GO_DEPS` where we currently use `cmd/podman/varlink/ioprojectatomicpodman.go` as a dependency.

Reply:
can you provide an example?  i like where you are headed but my Make-fu sucks ... alternatively and perhaps even better, how would you feel about pushing this PR through (it does in fact work) and then having you make a superior PR that contains some of the great stuff you put here.

Reply:
> can you provide an example?

I've filed a squash commit as baude/libpod#1.  Squash that commit in if you like it, and feel free to mess with the commit message suggestions.